databricks_cli/unity_catalog/delta_sharing_cli.py

```diff
@@ -141,20 +146,24 @@ def shared_data_object(name):
 @profile_option
 @eat_exceptions
 @provide_api_client
-def update_share_cli(api_client, name, add_table, remove_table, json_file, json):
+def update_share_cli(api_client, name, new_name, comment, owner,
+                     add_table, remove_table, json_file, json):
     """
     Update a share.
 
     The public specification for the JSON request is in development.
     """
-    if len(add_table) > 0 or len(remove_table) > 0:
+    if ((new_name is not None) or (comment is not None) or (owner is not None) or
+        len(add_table) > 0 or len(remove_table) > 0):
+        if (json_file is not None) or (json is not None):
+            raise ValueError('Cannot specify JSON if any other update flags are specified')
         updates = []
         for a in add_table:
             updates.append({'action': 'ADD', 'data_object': shared_data_object(a)})
         for r in remove_table:
             updates.append({'action': 'REMOVE', 'data_object': shared_data_object(r)})
-        d = {'updates': updates}
-        share_json = UnityCatalogApi(api_client).update_share(name, d)
+        data = {'name': new_name, 'comment': comment, 'owner': owner, 'updates': updates}
```

Comment:
I'm wondering if we should filter out keys that have value `None`.  Probably not needed now, since the server won't see those fields defined. We'll need to revisit this once the UC 'update' APIs allow clearing fields (with FieldMasks or whatever).

Reply:
Yeah, we can do that later, since all the other APIs do something similar along the lines.